File: Capitulo2- Conceitos/ConceitosFundamentais/ConceitosFundamentais.Funcional/PeriodoDeTempo.fs
module PeriodoDeTempo
open System

type Periodo = {DataInicial:DateTime; DataFinal:DateTime}

let adicionaDias periodo dias =
    { 
      DataInicial = periodo.DataInicial.AddDays(dias) 
      ; DataFinal=periodo.DataFinal.AddDays(dias)
    }




let verificarSeDataEstaEntreOPeriodo periodo (dataParaTestar:DateTime) = 
    dataParaTestar.CompareTo(periodo.DataInicial) >= 0 && 
    dataParaTestar.CompareTo(periodo.DataFinal) <= 0    

File: Capitulo2- Conceitos/ConceitosFundamentais/ConceitosFundamentais.NetFramework/PeriodoDeTempo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConceitosFundamentais
{
    public class PeriodoDeTempo
    {
        public DateTime DataInicial { get; set; }
        public DateTime DataFinal { get; set; }

        public bool VerificarSeDataEstaEntreOPeriodo
            (DateTime dataParaTestar)
        {
            return dataParaTestar.CompareTo(DataInicial) >= 0
                    && dataParaTestar.CompareTo(DataFinal) <= 0;
        }

        public void AdicionarDias(int dias)
        {
            DataInicial.AddDays(dias);
            DataFinal.AddDays(dias);
        }
    }
}
